src/rules/noUseBeforeDeclareRule.ts

```diff
@@ -79,7 +79,8 @@ function walk(ctx: Lint.WalkContext<void>, checker: ts.TypeChecker): void {
         const declaredBefore = declarations.some((decl) => {
             switch (decl.kind) {
                 case ts.SyntaxKind.FunctionDeclaration:
-                    // Functions may be declared later.
+                case ts.SyntaxKind.ClassDeclaration:
+                    // Functions and classes may be declared later.
```

Comment:
This is wrong. Classes are not hoisted and can therefore not be used before their declaration.

Reply:
Yeah, you're right of course. Can you take down the 'Accepting PRs' label from this issue: #3011 ? Or close it since it's not a valid request?